docs/usage/configuration/config_documentation.md

```diff
@@ -533,6 +548,21 @@ listeners:
     bind_addresses: ['::1', '127.0.0.1']
     type: manhole
 ```
+Example configuration #3:
+```yaml
+listeners:
+  # UNIX socket listener: for when Synapse is behind a reverse proxy that can utilise such.
```

Comment:
```suggestion
  # Unix socket listener: for when Synapse is behind a reverse proxy and (TODO: better reason).
```

Reply:
I don't think there is a better reason. It will only work if there is a way to proxy incoming requests to a socket file. I think my American vernicular was sticking it's nose in here. Probably could just remove the `that can utilise such`.

Reply:
I think they were mentioned in the original issue:
> It would be useful for people who don't want to open a localhost TCP port when they have a proxy in front of synapse.

> +1 to this request. It's really common practice for app servers like gunicorn, to skip local networking and just use sockets between the frontend (like nginx) and the app server, e.g. http://docs.gunicorn.org/en/stable/deploy.html

> This would be cool. I am not really a fan of using unprivileged TCP ports for loopbacks. Any unprivileged local user could seize the port if the server shuts down for a while. This issue could perhaps therefore be tagged with the "security" tag?

Reply:
Furthermore, as I understand it, there is slightly less overhead with Unix socket versus TCP sockets. A lot of the baggage that was built into TCP/IP over the decades to handle a bunch of issues(bufferbloat, blackholes, misconfigured anything, etc) that had to do with connectivity, Unix sockets just....don't have.

Reply:
Thanks for the context! Perhaps this:

```suggestion
  # Unix socket listener: Ideal for Synapse deployments behind a reverse proxy, offering lightweight interprocess communication without TCP/IP overhead, avoid port conflicts, and providing enhanced security through system file permissions.
```